internal/rbd/encryption.go

```diff
@@ -463,6 +465,34 @@ func (rv *rbdVolume) RotateEncryptionKey(ctx context.Context) error {
 		return errors.New("key rotation not supported for unencrypted device")
 	}
 
+	// Call open Ioctx to create a new ioctx object
+	// if the obj already exists, no error is returned
+	err = rv.openIoctx()
+	if err != nil {
+		log.ErrorLogMsg("failed to open ioctx, err: %v", err)
+
+		return fmt.Errorf("failed to open ioctx, err: %w", err)
+	}
+
+	// Lock params
+	lockName := rv.VolID + "-mutexlock"
+	lockDesc := "Key rotation mutex lock for " + rv.VolID
+	lockDuration := 3 * time.Minute
+	lockCookie := rv.VolID + "-enc-key-rotate"
+	var flags byte = 0
+
+	// Acquire the exclusive lock based on vol id
+	lck := lock.NewLock(rv.ioctx, rv.VolID, lockName, lockCookie, lockDesc, lockDuration, &flags)
+	err = lck.LockExclusive(ctx)
+	//nolint:errcheck // No need to log here again
```

Comment:
log the error which we got during taking lock

Reply:
The said error is logged at line 488. This comment is to suppress linter's warning due to not checking the error returned by Unlock. We do nothing in case an error is returned while unlocking except to log. Logging is already done in Unlock method.